### PR TITLE
feat: Support SubscribeTopicEvents API

### PR DESCRIPTION
**Pull request by CrazyHZM**
Github Action Bot detected the issue body's language is not English, translate it automatically
---

**What this PR does**:

see https://github.com/mosn/layotto/issues/1074

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```
<details><summary>原文</summary>

**What this PR does**:

see https://github.com/mosn/layotto/issues/1074

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```</details>
<!--MD5:059efee4dbfad356df5d8122cd78c695:MD5-->

### CODE REVIEW

File: pkg/grpc/default_api/api.go [hunk 1]
```diff
@@ -147,7 +150,9 @@ func NewAPI(
 		sendToOutputBindingFn:    sendToOutputBindingFn,
 		secretStores:             secretStores,
 		json:                     jsoniter.ConfigFastest,
+		//closeCh:                  make(chan struct{}),
```
[the remark this answers is not in the document]
Comment (CrazyHZM) [reply pkg/grpc/default_api/api.go]: 暂时先去掉

File: pkg/grpc/default_api/api.go [hunk 2]
```diff
@@ -86,8 +86,10 @@ type api struct {
 	// app callback
 	AppCallbackConn   *grpc.ClientConn
 	topicPerComponent map[string]TopicSubscriptions
+	streamer          *streamer
 	// json
 	json jsoniter.API
+	wg   sync.WaitGroup
```
Comment (wenxuwan) [inline pkg/grpc/default_api/api.go:92]: 这个a.wg.Add了两次，done了两次，wg.wait()没有，没什么用吧？

File: pkg/grpc/default_api/api_subscribe.go [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+	log.DefaultLogger.Infof("Subscribing to pubsub '%s' topic '%s'", req.GetPubsubName(), req.GetTopic())
+	s.lock.Unlock()
+
+	defer func() {
+		s.lock.Lock()
+		delete(s.subscribers, key)
+		s.lock.Unlock()
+	}()
+
+	var wg sync.WaitGroup
```
Comment (wenxuwan) [inline pkg/grpc/default_api/api_subscribe.go:169]: 这个地方没必要起个wg吧，下面的for已经是个循环了，起个goroutine没必要在外面等待，本身也不会做任何的收尾

File: pkg/grpc/default_api/api_subscribe.go [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+		return fmt.Errorf("no streamer subscribed to pubsub %q topic %q", pubsubName, msg.Topic)
+	}
+
+	envelope, cloudEvent, _ := a.envelopeFromSubscriptionMessage(ctx, msg)
+
+	ch, defFn := conn.registerPublishResponse(envelope.GetId())
+	defer defFn()
+
+	conn.streamLock.Lock()
+	err := conn.stream.Send(&runtimev1pb.SubscribeTopicEventsResponse{
```
Comment (wenxuwan) [inline pkg/grpc/default_api/api_subscribe.go:215]: 这里如果发失败了，ctx没有cancel或者超时下面的select就阻塞了

**Review by wenxuwan — CHANGES_REQUESTED** [review]

**Review by wenxuwan — APPROVED** [review]

**Pull request merged**